docs/core/plugins/sieve.md

```diff
@@ -46,128 +58,191 @@ protocol lmtp {
 }
 ```
 
-## Script Locations
+## Script storage
 
-The Sieve interpreter can retrieve Sieve scripts from several types of
-locations.
+Sieve scripts are retrieved from a script storage. This can currently be the
+local filesystem, an LDAP database or any dict storage. Depending on the storage
+implementation, its type and its configuration, storages can contain one script,
+several scripts identified by name, and a series of scripts in a well-defined
+order to be executed in sequence.
 
-The default [file](#file) location type is a directory containing
-one or more Sieve script files with a symlink pointing to the active
-one.
-
-More complex setups can use other location types such as [LDAP](#ldap) or
-[dict](#dict) to fetch Sieve scripts from remote databases.
-
-All settings that specify the location of one or more Sieve scripts
-accept the following syntax:
+Script storages are configured in a named sieve_script block:
 
 ```
-<setting> = [<type>:]path[;<option>[=<value>][;...]]
+sieve_script personal {
+  path = ~/.dovecot.sieve
+}
```

Comment:
I've understood that this sieve_script block isn't really necessary, right? There is a default script anyway being executed? Should explain how that works here.

Reply:
If no personal storage is defined, auto-detection will be attempted. If no storage is found, Sieve processing will be skipped. Script from default storage is only executed when a personal storage is available, yet no script is active.

Reply:
explained for personal storage type